src/core/thread/key_manager.cpp

```diff
@@ -443,4 +526,137 @@ void KeyManager::HandleKeyRotationTimer(void)
     }
 }
 
+NetworkKey KeyManager::GetNetworkKey(void)
+{
+    NetworkKey networkKey;
+
+#if OPENTHREAD_CONFIG_PLATFORM_KEY_REFERENCES_ENABLE
+    Error  error   = kErrorNone;
+    size_t mKeyLen = 0;
+
+    error = otPlatCryptoExportKey(mNetworkKeyRef, networkKey.m8, sizeof(networkKey.m8), &mKeyLen);
+
+    OT_ASSERT(error == kErrorNone);
+#else
+    networkKey = mNetworkKey;
+#endif
+
+    return networkKey;
+}
+
+const Pskc KeyManager::GetPskc(void) const
+{
+    Pskc pskc;
+
+#if OPENTHREAD_CONFIG_PLATFORM_KEY_REFERENCES_ENABLE
+    Error  error   = kErrorNone;
+    size_t mKeyLen = 0;
+
+    error = otPlatCryptoExportKey(mPskcRef, pskc.m8, sizeof(pskc.m8), &mKeyLen);
+
+    OT_ASSERT(error == kErrorNone);
+#else
+    pskc = mPskc;
+#endif
+
+    return pskc;
+}
+
+#if OPENTHREAD_CONFIG_PLATFORM_KEY_REFERENCES_ENABLE
+
+Error KeyManager::StoreNetworkKey(bool aOverWriteExisting)
+{
+    Error       error = kErrorNone;
+    otMacKeyRef keyRef;
+
+    keyRef = kNetworkKeyPsaItsOffset;
+
+    if (!aOverWriteExisting)
+    {
+        otCryptoKeyAttributes mKeyAttributes;
+
+        error = otPlatCryptoGetKeyAttributes(keyRef, &mKeyAttributes);
+        // We will be able to retrieve the key_attributes only if there is
+        // already a network key stored in ITS. If stored, and we are not
+        // overwriting the existing key, return without doing anything.
+        if (error == OT_ERROR_NONE)
+        {
+            ExitNow();
+        }
+    }
+
+    CheckAndDestroyStoredKey(keyRef);
```

Comment:
@hemanth-silabs just double-checking, I think here we want to `Destroy()` the `mNetworkKeyRef` and not the `keyRef` that we set above to `kNetworkKeyPsaItsOffset`. I think this is a bug, but want to check with you to make sure I am not missing something here. 
Will fix/change it in the next commit I will push.

Reply:
Reading the code, I think the idea is that we are hard-coding the `KeyRef` to be used for the NetworkKey and Pskc to be fixed? In my understanding correct? 


Does it mean that when we call `otPlatCryptoImportKey()` and pass a `otCryptoKeyRef *` assigned to some value, the caller is already specifying the `KeyRef` vlaue to be used ? Is this the expected behavior when `OT_CRYPTO_KEY_STORAGE_PERSISTENT` is used? Is it then expected that `otPlatCryptoImportKey()` does not change the passed-in `otCryptoKeyRef *`?

@hemanth-silabs  can you please help shed some light on this? Thanks.

Reply:
@abtink You are correct. `keyRef` is a in/out parameter in `otPlatCryptoImportKey`. Depending on the persistence, PSA will either generate a new key_id or use the one provided by the caller. For persistent keys PSA will not modify the passed `otCryptoKeyRef *`

Reply:
Got it. Thanks. I suggest we explain this expected behavior and add this in the docs for the `otPlatCryptoImportKey()` API. I think it will help make it clear to someone reading the platform API how it is expected to behave. 

Or another idea (which may be better) is to break the `otPlatCryptoImportKey()` into two APIs: 
- `otPlatCryptoImportVolatileKey()` which the API itself assigns `KeyRef` and return it in a passed arg `otCryptoKeyRef *`.
- `otPlatCryptoImportPersistantKey()` which expects the `KeyRef` to be provided by caller (won't need a pointer and we just pass it an `otCryptoKeyRef` value).

The advantages I see with doing this are it helps make the distinctions very clear, and avoid confusion of how to treat the parameters (input vs output) and potential mistake in implementation of platform API (in future on a new platforms).

Thoughts?